src/lib/operators/abstract_operator.hpp

```diff
@@ -127,16 +132,20 @@ class AbstractOperator : public std::enable_shared_from_this<AbstractOperator>,
   // clean up after execution (if it makes sense)
   virtual void _on_cleanup();
 
+  // override this if the Operator uses Expressions and set the parameters within them
+  virtual void _on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters);
```

Comment:
Do I see it right that the parameters also still set through the constructor? If so, is it possible to limit it to one place? Or else leave a comment when which way is used or should be used preferably...

Reply:
Nope, `set_parameters` is the only path through which ParameterExpressions/AllParameterVariants are assigned a value.